src/components/Messages.js

```diff
@@ -1,12 +1,78 @@
 import React from 'react'
+import {Component} from 'react'
 import { Button } from 'react-bootstrap';
+import ReactDOM from 'react-dom';
 
-function About() {
-    return <div className="content">
-        <h1>Messages page</h1>
-        <Button> Inbox </Button> 
-        <Button> Sent </Button>
-      </div>;
+  class Messages extends React.Component {
+    constructor(props) {
+      super(props);
+      this.handleInboxClick = this.handleInboxClick.bind(this);
+      this.handleBackClick = this.handleBackClick.bind(this);
+      this.state = {isLoggedIn: false};
+    }
+  
+    handleInboxClick() {
+      this.setState({isLoggedIn: true});
+    }
+  
+    handleBackClick() {
+      this.setState({isLoggedIn: false});
+    }
+  
+    render() {
+      const isLoggedIn = this.state.isLoggedIn;
+      let button;
+  
+      if (isLoggedIn) {
+        button = <BackButton onClick={this.handleBackClick} />;
+      } else {
+        button = <InboxButton onClick={this.handleInboxClick} />;
+      }
+  
+      return (
+        <div>
+          <Greeting isLoggedIn={isLoggedIn} />
+          {button}
+        </div>
+      );
+    }
+  }
+  
+  function UserGreeting(props) {
+    return <h1>Welcome to Inbox Page!</h1>;
+  }
+  
+  function GuestGreeting(props) {
+    return <h1>Welcome to Messages Page!</h1>;
+  }
+  
+  function Greeting(props) {
```

Comment:
The naming here also doesn't seem very clear.  Why is this called "Greeting" when it seems to be for showing the inbox content?  "InboxContent" might be a bit more accurate to what it's supposed to do.

Also, the reason I'm nitpicking a bit about the naming is because if something is named confusingly then you and other developers might have a hard time working on and improving your code.  It's important to consider good names when developing for everyone's sake.